.github/workflows/ci.yml

```diff
@@ -193,51 +186,31 @@ jobs:
 
       - id: push_cassettes
         name: Push updated cassettes
+        if: always() && (startsWith(github.event_name, 'pull_request') || success())
         run: |
           if [ "${{ startsWith(github.event_name, 'pull_request') }}" = "true" ]; then
             is_pull_request=true
             cassette_branch="${{ github.event.pull_request.user.login }}-${{ github.event.pull_request.head.ref }}"
-            cassette_source_branch="${{ steps.checkout_cassettes.outputs.cassette_branch }}"
-            base_branch="${{ github.event.pull_request.base.ref }}"
           else
-            current_branch=$(echo ${{ github.ref }} | sed -e "s/refs\/heads\///g")
-            cassette_branch=$current_branch
+            cassette_branch=$(echo ${{ github.ref }} | sed -e "s/refs\/heads\///g")
           fi
 
           cd tests/Auto-GPT-test-cassettes
```

Comment:
here I deleted a lot of code because things are more straightforward, thanks to the previous steps.